mathjax3-ts/output/chtml/CssStyles.ts

```diff
@@ -0,0 +1,113 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2017 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview  Implements the CssStyles class for handling stylesheets
+ *
+ * @author dpvc@mathjax.org (Davide Cervone)
+ */
+
+/*
+ * The data for a selector
+ */
+export type StyleData = {
+    [property: string]: string | number;
+};
+
+/*
+ * A list of selectors and their data (basically a stylesheet)
+ */
+export type StyleList = {
+    [selector: string]: StyleData;
+};
+
+/******************************************************************************/
+/*
+ * The CssStyles class (for managing a collection oof CSS style definitions
+ */
+
+export class CssStyles {
+    /*
+     * The styles as they currently stand
+     */
+    protected styles: StyleList = {};
+
+    /*
+     * @return{string}  The styles as a CSS string
+     */
+    get cssText() {
+        return this.getStyleString();
+    }
+
+    /*
+     * @param{StyleList} styles  The initial styles to use, if any
+     * @constructor
+     */
+    constructor(styles: StyleList = null) {
+        this.addStyles(styles);
+    }
+
+    /*
+     * @param{StyleList} styles  The styles to combine with the existing ones
+     */
+    public addStyles(styles: StyleList) {
+        if (!styles) return;
+        for (const style of Object.keys(styles)) {
```

Comment:
use for ... in.

Reply:
That may enumerate things that we don't want.  For example, if someone adds a method to the `Object.prototype`, `for ... in` would list that as one of the keys.  So we would have to add and `if-then` for `styles.hasOwnProperty()` to make sure it was a top-level key (that would increase the nesting level, and I know you don't like that).  `Object.keys()` does that already.  I'm not sure what is gained by going to a more complicated (and deeply nested) construct.